plugins/inputs/kernel/kernel.go

```diff
@@ -98,31 +111,87 @@ func (k *Kernel) Gather(acc telegraf.Accumulator) error {
 		}
 	}
 
+	if k.optCollect["ksm"] {
+		stats := map[string]interface{}{
+			"full_scans":                         0,
+			"max_page_sharing":                   0,
+			"merge_across_nodes":                 0,
+			"pages_shared":                       0,
+			"pages_sharing":                      0,
+			"pages_to_scan":                      0,
+			"pages_unshared":                     0,
+			"pages_volatile":                     0,
+			"run":                                0,
+			"sleep_millisecs":                    0,
+			"stable_node_chains":                 0,
+			"stable_node_chains_prune_millisecs": 0,
+			"stable_node_dups":                   0,
+			"use_zero_pages":                     0,
+		}
+		// these exist in very recent Linux versions only, but useful to include if there.
+		extraStats := map[string]interface{}{
+			"general_profit": 0,
+		}
```

Comment:
Was there a purpose for using a map versus a slice of strings for `stats` and `extraStats`? I only ask as I assumed based on the maps that you were storing the results in the map versus updating `fields`.

Reply:
Heh, used a map mostly out of habit in case I wanted to pull specific keys when I was getting this together this initially in the `kernel_ksm` plugin, and then I never really looked back at it.

Given how this eventually turned out - ie. adding directly to the fields map, it makes sense, I'll replace with slices.